ezidapp/models/minter.py

```diff
@@ -0,0 +1,68 @@
+#  Copyright©2023, Regents of the University of California
+#  http://creativecommons.org/licenses/BSD
+
+"""Object Relational Mapper (ORM) models for the minters
+"""
+import time
+import logging
+
+import django.db.models
+import django.core.serializers.json
+import django.core.validators
+from ezidapp.models  import validation
+import impl.util
+
+logger = logging.getLogger(__name__)
+
+def default_time():
+    return int(time.time())
+
+class Minter(django.db.models.Model):
+    # Class for the Minters. 
+    # Minters are uniquely identified by a prefix/shoulder;
+    # Each minter maintains its own state which is stored in the minterState field.
+
+    # The shoulder itself, qualified and normalized, e.g., "ark:/12345/"
+    # or "doi:10.1234/FOO".
+    prefix = django.db.models.CharField(
+        max_length=impl.util.maxIdentifierLength,
+        unique=True,
+        validators=[validation.shoulder],
+    )
+
+    # minter state in JSON
+    minterState = django.db.models.JSONField(
+        encoder=django.core.serializers.json.DjangoJSONEncoder,
+        default=dict,
+    )
+    
+    # The time the identifier was created as a Unix timestamp. If not
+    # specified, the current time is used.
+    createTime = django.db.models.IntegerField(
+        default=default_time(),
```

Comment:
Here default is set to the value when default_time is executed, which is when the python interpreter loads the file, not when the class is instantiated. Should probably use the Django ORM mechanisms for auto timestamps, e.g. `auto_now_add` and `auto_now`.

Reply:
It would be a callable if it was `default=default_time`, but as it is, you get a static value evaluated when the module is loaded.

Reply:
Since the createTime and updateTime are int, we cannot use auto timestamps. How about just set the default value to zero and have business logic to handle the real values?

Reply:
default=default_time (i.e. without parentheses) worked! Thank you Dave.